include/functions_routing.inc.php

```diff
@@ -47,7 +47,7 @@ function locateHiddenVariables($_args) {
         } elseif ($v[0] == 'A') { /* Author */
             $url_author = substr($v, 1);
             if (is_numeric($url_author)) {
-                $serendipity['GET']['viewAuthor'] = $_GET['viewAuthor'] = (int)$url_author;
```

Comment:
I believe there was a reason to also have ?viewAuthor=XXX instead of only ?serendipity[viewAuthor] at some place. Did you check that also possible other plugins might reference this $_GET['viewAuthor'] variable?

Reply:
As far as I could determine, the RSS feed issues the requests outside of the [serendipity] array, i think this is because of the RSS interface, and because of that I left the serveFeed() function untouched. viewAuthor is not set directly into GET from any other function or plugin. I did my little purge here because some of the 'lang' variables were used in $serenditpity['GET'] and $_GET at the same time and written to which really created a mess. Ideally, all GET should be moved into a [serendipity] array, but there are usages in admin/plugins.inc.php, exit.php and comment.php where i couldn't identify the source of and because of that i didn't touch them.

There is also the occurrence of $_GET['p'] in serveEntry() which I don't know what that is for. That means you can get an entry with `/index.php/?p=33`   where p is the entry id. But basically, it would work even with `/index.php/?33` .

Reply:
OK, understood. Well, then just lets try to remove that viewAuthor and see if it breaks things ;)

That "?p" variable is Wordpress compatibility simply, for people who moved from one system to the other by keeping entry IDs and without needing specific rewrite rules for compatibility.